react/react-ecosystem/type-checking-with-proptypes.md

```diff
@@ -0,0 +1,100 @@
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Setting up PropTypes.
+- Using common PropTypes features.
+
+### Getting Started
+
+To start using PropTypes in our react projects, we first need to install the corresponding library. We can do that with ~~~npm~~~. In your react project run the following command:
+
+~~~
+npm install --save prop-types
+~~~
+
+Next, we want to import the PropTypes package in the component whose props we want to validate.
+
+~~~javascript
+import PropTypes from 'prop-types';
+~~~
+
+### Using propTypes
+
+Here is a very basic example of how we would use it in a simple component that renders out a name prop.
+
+~~~javascript
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const RenderName = (props) => {
+  return <div>{props.name}</div>;
+};
+
+RenderName.propTypes = {
+  name: PropTypes.string,
+};
+
+export default RenderName;
+~~~
+
+In this example, the component RenderName expects to receive a prop called name which is a string. If this prop is not passed, or if it is not a string, a warning will be displayed.
+
+### Using defaultProps
+
+Another cool thing we can do with PropTypes is passing in default props:
+
+~~~javascript
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const RenderName = (props) => {
+  return <div>{props.name}</div>;
+};
+
+RenderName.propTypes = {
+  name: PropTypes.string,
+};
+
+RenderName.defaultProps = {
+  name: 'Zach',
+};
+
+export default RenderName;
+~~~
+
+In this example, with the help of the defaultProps property we are defining a default value for the `name` prop. This way, if the `RenderName` component is called without passing in the `name` prop, it will default to "Zach". When you do pass in props, they will take precedence over the default props.
+
+### What about TypeScript?
+
+Now is also a good time to mention [TypeScript](typescriptlang.org) - a strongly typed language that builds on JavaScript. We don't cover it specifically, but it's worth learning about it if you'd like more safety while writing your code.
+
+If you go in the direction of learning TypeScript, our recommendation would be picking up a previous project and refactoring the components one by one to TypeScript. Learn by doing!
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1. Read through the [PropTypes documentation](https://reactjs.org/docs/typechecking-with-proptypes.html). It shows all of the types you can specify and some other useful things that can be done with it!
+2. Read [this article](https://blog.logrocket.com/validate-react-props-proptypes/) for a more in-depth look into the benefits and use cases of PropTypes.
```

Comment:
Love this resource. Perhaps we can improve link text and provide a sneak peek of sorts?

```suggestion
2. You can even set up custom validators in PropTypes. Read [this article on using PropTypes on LogRocket](https://blog.logrocket.com/validate-react-props-proptypes/) for a more in-depth look into the benefits and use cases of PropTypes.
```

Reply:
could you elaborate more on the sneak peek you're referring to? I'm not sure that we have historically given sneak peeks in this section. Were you referring to the custom validators part or were you thinking something more along the lines of screenshots? 

sorry for the confusion on my part !

Reply:
> Were you referring to the custom validators part

yup, that's definitely what I meant